src/library/scala/MatchError.scala

```diff
@@ -22,7 +22,7 @@ final class MatchError(@transient obj: Any) extends RuntimeException {
   /** There's no reason we need to call toString eagerly,
    *  so defer it until getMessage is called or object is serialized
    */
-  private lazy val objString = {
+  private[this] lazy val objString = {
```

Comment:
For lazy vals it doesn't change anything in the bytecode, but we can keep them in anyway.